text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,1650 @@
+```
+$ cdk init app --language=typescript --lib-version 2.100.0
+```
+
+To use a custom stack name for your project (built-in templates only), add the `--stack-name` flag to your command:
+```
+$ cdk init app --language=typescript --stack-name MyCustomStack
+```
+
+
+#### **CDK Template Registry**
+
+The CDK CLI includes a vetted registry of public AWS templates that provide pre-configured infrastructure patterns for specific use cases. To see all available templates to initialize a CDK project with, including those in the public registry, run `cdk init --list`:
+
+```
+Available templates:
+* app: CDK Application Template
+   └─ cdk init app --language=[typescript|javascript|python|java|csharp|fsharp|go]
+* lib: CDK Construct Library Template
+   └─ cdk init lib --language=typescript
+* sample-app: Example CDK Application with constructs
+   └─ cdk init sample-app --language=[typescript|javascript|python|java|csharp|fsharp|go]
+
+Public template registry:
+┌────────────────┬─────────────────────────────────────┬───────────────────────┬────────────────────────────────────────────────────────────────────┐
+│ Name           │ Description                         │ Author                │ Usage                                                              │
+├────────────────┼─────────────────────────────────────┼───────────────────────┼────────────────────────────────────────────────────────────────────┤
+│sample-git-repo │ Sample public GitHub repository with│ @rohang9000           │--from-github=rohang9000/sample-git-repo --template-path my-template│
+│                │ a custom template                   │                       │--language=[python|typescript]                                      │
+└────────────────┴─────────────────────────────────────┴───────────────────────┴────────────────────────────────────────────────────────────────────┘
+```
+
+
+
+### Authoring Custom CDK Templates
+
+The requirements for a custom template are that it contains a CDK supported language subdirectory and at least one file of the same type inside that subdirectory. Reference the appendix for example app template implementations in all CDK Languages.
+
+#### Custom Template Schema
+
+```
+my-custom-app-template/
+└── [language-name]/             # REQUIRED: CDK language subdirectory 
+    └── [file-name].[ext]        # REQUIRED: At least one file matching the language type in language subdirectory
+```
+
+##### Language-Specific Substitutions:
+
+* [language-name]: csharp, fsharp, go, java, javascript, python, typescript
+* [ext]: .cs, .fs, .go, .java, .js, .py, .ts
+
+
+#### Testing Your Custom Templates Locally
+
+After authoring your custom templates, it can be helpful to test that `cdk init` runs successfully before publishing them to a Git repository or NPM package. Projects can be initialized from the authored template stored locally on your filesystem:
```

Comment:
I would say that running the command successfully is not the measure of success, but rather a step as part of self-review for any template author.

i.e. an author should run `cdk init` to validate that the output of their custom template is what they expect.